qiling/extensions/pipe.py

```diff
@@ -48,11 +48,28 @@ def writable(self) -> bool:
     def readable(self) -> bool:
         return True
 
+    def seek(self, offset: int, origin: int) -> int:
+        # Imitate os.lseek
+        raise OSError("Illega Seek")
+
+    def lseek(self, offset: int, origin: int) -> int:
```

Comment:
`lseek` is not part of the `TextIO` interface, suggesting to remove this and let the caller use `seek` (after checking that it is indeed `seekable`)

Reply:
There are too many things that have to be checked if we do this in this way, for instance, `readable`, `writable` etc. My idea is just to keep the same behavior as `sys.stdin`.

Reply:
I agree: each syscall should check that it applies to the target fd before it does anything.
`write` should check first fd is writeable, `read` should check first fd is readable, and so on.. there are specific errors just for that.

That implementation would be much closer to the spec.

Reply:
Fine, now we don't need to check since an exception will be raised like other streams.